packages/react/src/number-field/root/NumberFieldRoot.tsx

```diff
@@ -542,7 +545,10 @@ export const NumberFieldRoot = React.forwardRef(function NumberFieldRoot(
         value={value ?? ''}
         min={min}
         max={max}
-        step={step}
+        // stepMismatch validation is broken unless an explicit `min` is added.
```

Comment:
Let's add a note in the props descriptions that min needs to be provided if the step prop is used.